application/managers/http/manager.py

```diff
@@ -0,0 +1,76 @@
+"""
+Manager for http business logic.
+"""
+
+
+from fastapi import Depends
+
+from constants.events import EventCategory
+from managers.events import EventManager
+from managers.events import get_event_manager
+from managers.kubernetes import K8sManager
+from managers.organizations.manager import OrganizationManager
+from managers.organizations.manager import get_organization_manager
+from models.organization import Organization
+from schemas.events import EventSchema
+from services.helm.facade import HelmService
+import httpx
+
+
+
+class HttpManager:
+    """
+    Manager to manipulate Http requests.
+    """
+    organization_manager: OrganizationManager
+
+    def __init__(self, organization_manager: OrganizationManager, event_manager: EventManager):
+        self.organization_manager = organization_manager
+        self.event_manager = event_manager
+
+
+    ############################################################################
+    # Http Requests
+    ############################################################################
+
+
+    async def http_request(
+        self, organization: Organization, context_name: str, component_name: str, url: str, method: str,
+        parameters: dict, headers: dict = None, dry_run: bool = False
+    ) -> str:
+        """
+        Send an HTTP request to the specified URL.
+        """
+        # Convert to a dict if a list of dicts provided
+        if isinstance(parameters, list):
+            parameters = {k: v for d in parameters for k, v in d.items()}
+
+        # Create an HTTP client
+        async with httpx.AsyncClient() as client:
+            # Prepare the request
+            if method.lower() == 'get':
+                response = await client.get(url, params=parameters, headers=headers)
+            elif method.lower() == 'post':
+                response = await client.post(url, params=parameters, headers=headers)
+            elif method.lower() == 'put':
+                response = await client.put(url, params=parameters, headers=headers)
+            elif method.lower() == 'delete':
+                response = await client.delete(url, params=parameters, headers=headers)
+            elif method.lower() == 'patch':
+                response = await client.patch(url, params=parameters, headers=headers)
+            else:
+                raise ValueError(f"Unsupported method: {method}")
```

Comment:
I think you need to create new exception module with exception related to http manager in `application/exceptions/<...>` and not raise to common `ValueError`.

Also I think you need to intercept exceptions from `httpx` and convert it to http manager's exceptions. Don't shift the internal implementation guts on end user.